### DISCUSSION TITLE
"Connection closed while SSL/TLS handshake was in progress" Failing after migrating the service in Azure

### DISCUSSION THREAD

**Discussion by Jagkalai10** [relayrides/pushy#1092] [Q&A]
I see similar error after migrating the service in Azure. Firewall opened for outbound to api.push.apple.com, not sure why we don't see any traffic after DNS is resolved. Can you please advise if there any resolution on this?

nioEventLoopGroup-2-1 | c.t.p.a.ApnsChannelFactory | | | Tried to mark promise as "failed," but it was already done. java.nio.channels.ClosedChannelException: null at io.netty.handler.ssl.SslHandler.channelInactive(SslHandler.java:1150) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:305) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281) at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:274) at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:301) at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281) at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901) at io.netty.channel.AbstractChannel$AbstractUnsafe$7.run(AbstractChannel.java:813) at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566) at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) at java.base/java.lang.Thread.run(Thread.java:829) Suppressed: io.netty.handler.ssl.StacklessSSLHandshakeException: Connection closed while SSL/TLS handshake was in progress at io.netty.handler.ssl.SslHandler.channelInactive(Unknown Source)

2024-11-20 16:42:50,640 | DEBUG | nioEventLoopGroup-2-1 | i.n.r.dns.DnsNameResolver | | | [id: 0xae756d38] RECEIVED: UDP [7263: /172.25.0.10:53], DatagramDnsResponse(from: /172.25.0.10:53, to: /0:0:0:0:0:0:0:0:47575, id: 7263, QUERY(0), NoError(0), RD RA) DefaultDnsQuestion(api.push.apple.com. IN A) DefaultDnsRawRecord(api.push.apple.com. 19 IN CNAME 34B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(api-vs.push-apple.com.akadns.net. 19 IN A 4B) DefaultDnsRawRecord(OPT flags:0 udp:4096 0B)

**Comment by jchambers** [comment]
In short: I don't know. It sounds like this problem is somewhere in your specific configuration.

What have you already tried to solve this problem on your own? What happens if you `curl https://api.push.apple.com` from your server?

**Reply by Jagkalai10** [reply]
curl or openssl to this api works from the server. But from the spring boot app it fails in ApnsClient in sendNotification method. It fails in ApnsChannelPool while in acquire() method, its not an this.executor.inEventLoop() so it jumps in else block and throws "Channel closed before HTTP/2 preface completed" in this.executor.submit block.
 Please advise if there any specific netty exclusion to be done for the app deployed in AKS
